docs/tech_specs/configuration_backend_approach.md

```diff
@@ -0,0 +1,107 @@
+- Fetch dependencies and start referencing Screens code
+
+### Benefits
+
+Setting up the shared code as a separate library and importing it as a package into Screenplay is a less intrusive solution since Screens and Screenplay will remain as separate projects. It also encourages a more modular structure and clearer separation of concerns. This approach is also arguably a more intuitive way of sharing code between projects and would therefore be less likely to be disruptive to the team and it’s workflow. A library also offers more flexibility since upgrades can be managed independently and an update to the shared code won’t trigger an automatic update to all repos to reference it.
+
+### Drawbacks
+
+While a library does enable greater flexibility in terms of version management, it also requires extra effort to do the management.
+
+### References
+
+https://hexdocs.pm/elixir/main/library-guidelines.html
+
+## 3. Git Dependency
+
+### Description
+
+Elixir also supports adding a GitHub repository directly as a dependency to your project. You can also pin the dependency to specific tags/commits/branches as well.
+
+### High-level steps to get there
+
+- Add the git dependency to the deps in Screenplay’s `mix.exs` file
+- Fetch dependencies and start referencing Screens code
+
+### Benefits
+
+This method is the least intrusive and easiest as it requires a trivial amount of upfront work to enable Screenplay to reference the code from Screens. It also requires no changes to the Screens application since we don’t need to move the shared code anywhere. We also have some of the same flexibility as the library approach since we can pin Screenplay to a specific version of Screens.
+
+### Drawbacks
+
+One problem with this approach is that there are no enforceable boundaries. In other words, there is nothing preventing Screenplay from referencing code anywhere in Screens.
+
+### References
+
+https://hexdocs.pm/mix/1.12.3/Mix.Tasks.Deps.html
+
+# Recommendation
+
+The approach I recommend is number 2, the library approach. This approach requires some upfront work to set up like the umbrella approach, but is less intrusive to our team’s workflow. Umbrella projects are more suited for cases where there are many, tightly integrated applications, which is not the case with Screens and Screenplay. In addition, setting up an umbrella project may backfire on us as it appears that it’s not a widely supported approach in the larger Elixir community and other teams within CTD have faced frustrations when using an umbrella, such as handy tools like Lcov being optimized for standard Elixir apps and not playing well with an umbrella setup.
+
+While the git dependency would be the lowest hanging fruit in achieving our goal, we will lack the ability to maintain any boundaries between Screenplay and Screens which isn’t what we want since the majority of Screens code will not be needed by Screenplay.
```

Comment:
Great write-up! My first look-over, I didn't have strong opinions. All options were well discussed, and it was hard to choose. This time, though, the git option catches my eye as a particular fave.

Is keeping boundaries here critical? I don't see an obvious downside to allowing Screenplay to be able to access the full Screens repo. The developers of the two repos have the same permission levels, so it isn't a concern of security. And while much of the Screens repo will not be useful to Screenplay, whatever isn't used should cause performance issues or dead weight, I think?

Reply:
Whoops, reading Jon's thoughts now!

Reply:
After reading Jon's notes and [looking around](https://elixirforum.com/t/importing-a-phoenix-app-as-a-dependency-in-another-phoenix-app-in-an-umbrella-project-please-any-advice/33144/3), I see why the first and third option would cause Screenplay to always fully spin up Screens, which wouldn't be ideal. So I like Jon's suggestion on recommendation 2: isolate the shared code, make a library, make it a GH dependency instead of publishing to hex.

Reply:
Functionally, I didn't think there were any obvious downsides either although some of @jzimbel-mbta 's comments I think point out that there could be some risks.

I think with the current team and everyone having a familiarity with both applications and the current work being done, the risk is low to none for doing a full git dependency. But if down the line a different org/team/individual ends up maintaining these applications without as much context, there's nothing to suggest to them that they shouldn't start using modules from `Screens` in `Screenplay` and vice versa.